src/main/java/org/marketdesignresearch/mechlib/mechanism/auctions/mlca/phases/ExactRandomQueryAuctionRoundBuilder.java

```diff
@@ -0,0 +1,44 @@
+package org.marketdesignresearch.mechlib.mechanism.auctions.mlca.phases;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.marketdesignresearch.mechlib.core.Outcome;
+import org.marketdesignresearch.mechlib.core.bid.bundle.BundleExactValueBids;
+import org.marketdesignresearch.mechlib.mechanism.auctions.Auction;
+import org.marketdesignresearch.mechlib.mechanism.auctions.AuctionRound;
+import org.marketdesignresearch.mechlib.mechanism.auctions.AuctionRoundBuilder;
+import org.marketdesignresearch.mechlib.mechanism.auctions.interactions.ExactValueQuery;
+import org.marketdesignresearch.mechlib.outcomerules.OutcomeRuleGenerator;
+import org.springframework.data.annotation.PersistenceConstructor;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED, onConstructor = @__({ @PersistenceConstructor }))
+public class ExactRandomQueryAuctionRoundBuilder extends AuctionRoundBuilder<BundleExactValueBids> {
+
+	@Getter
+	private final Map<UUID, ExactValueQuery> interactions;
+	
+	public ExactRandomQueryAuctionRoundBuilder(Auction<BundleExactValueBids> auction, Map<UUID, ExactValueQuery> interactions) {
+		super(auction);
+		this.interactions = interactions;
+	}
+
+	@Override
+	public AuctionRound<BundleExactValueBids> build() {
+		return new RandomQueryAuctionRound<>(this.getAuction(),
+				new BundleExactValueBids(this.interactions.entrySet().stream().collect(
+						Collectors.toMap(e -> this.getAuction().getBidder(e.getKey()), e -> e.getValue().getBid(),(e1,e2)->e1,LinkedHashMap::new))));
+	}
+
+	@Override
+	protected Outcome computeTemporaryResult(OutcomeRuleGenerator outcomeRuleGenerator) {
+		// TODO Auto-generated method stub

```

Comment:
Commenting such that this doesn't get lost (either properly implement it or throw an error that this is not yet implemented)

Reply:
Removed the method completly as it does not make sense in general to compute a temporary outcome without knowledge of past rounds. If temporary results are needed this should be rather added in the Auction itself in my opinion.